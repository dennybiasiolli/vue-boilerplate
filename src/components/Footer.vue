<template>
  <v-footer app>
    <v-spacer />
    &copy; {{ year }} - Denny Biasiolli
    <v-spacer />
  </v-footer>
</template>

<script>
export default {
  name: 'Footer',
  computed: Object.assign({
    year: () => new Date().getFullYear()
  })
}
</script>
