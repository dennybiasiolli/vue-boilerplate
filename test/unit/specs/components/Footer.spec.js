import { shallow } from '@vue/test-utils'
import Footer from '@/components/Footer'

describe('Footer', () => {
  test('should match snapshot', () => {
    const component = shallow(Footer)
    expect(component.element).toMatchSnapshot()
  })

  // Evaluate the results of functions in
  // the raw component options
  test('sets the correct components data', () => {
    expect(Footer.name).toBe('Footer')
    expect(Footer.computed.year()).toBe(new Date().getFullYear())
  })
})
