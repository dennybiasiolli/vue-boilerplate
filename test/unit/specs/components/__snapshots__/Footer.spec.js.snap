// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match snapshot 1`] = `
<v-footer-stub
  app=""
  height="32"
>
  <v-spacer-stub />
  
  Â© 2019 - Denny Biasiolli
  
  <v-spacer-stub />
</v-footer-stub>
`;
